src/main/java/com/uber/cadence/internal/common/WorkflowExecutionUtils.java

```diff
@@ -191,31 +194,46 @@ public static HistoryEvent getInstanceCloseEvent(
     long start = System.currentTimeMillis();
     HistoryEvent event;
     do {
+      if (timeout != 0 && System.currentTimeMillis() - start > unit.toMillis(timeout)) {
+        throw new TimeoutException(
+            "WorkflowId="
+                + workflowExecution.getWorkflowId()
+                + ", runId="
+                + workflowExecution.getRunId()
+                + ", timeout="
+                + timeout
+                + ", unit="
+                + unit);
+      }
+
       GetWorkflowExecutionHistoryRequest r = new GetWorkflowExecutionHistoryRequest();
       r.setDomain(domain);
       r.setExecution(workflowExecution);
       r.setHistoryEventFilterType(HistoryEventFilterType.CLOSE_EVENT);
       r.setNextPageToken(pageToken);
       r.setWaitForNewEvent(true);
+      r.setSkipArchival(true);
```

Comment:
When will `SkipArchival` be true? Is SkipArchival always true for long-poll?

Reply:
When user want to get archival history from CLI/UI. This method is shared by user and worker.